trin-core/src/portalnet/overlay.rs

```diff
@@ -490,6 +675,24 @@ impl<TContentKey: OverlayContentKey + Send, TMetric: Metric + Send>
     }
 }
 
+/// Randomly select log2 nodes. log2() is stable only for floats, that's why we cast it first to f32
+fn log2_random_enrs(enrs: Vec<Enr>) -> Vec<Enr> {
```

Comment:
My guess is that we might want to do something different later, but this seems like a fine starting point. IIRC, geth uses sqrt instead as a kind of preference for over-propagating. There also might be even more special cases when only connected to a small number of peers. We'll see as we experiment with the network...